src/back/mod.rs

```diff
@@ -20,16 +20,24 @@ impl crate::Expression {
     /// Note: we have to cache any expressions that depend on the control flow,
     /// or otherwise they may be moved into a non-uniform contol flow, accidentally.
     #[allow(dead_code)]
-    fn bake_ref_count(&self) -> usize {
+    fn bake_ref_count(&self, expressions: &crate::Arena<crate::Expression>) -> usize {
+        const OTHER_EXPR_BAKE_COUNT: usize = 2;
         match *self {
             // accesses are never cached, only loads are
             crate::Expression::Access { .. } | crate::Expression::AccessIndex { .. } => !0,
             // sampling may use the control flow, and image ops look better by themselves
             crate::Expression::ImageSample { .. } | crate::Expression::ImageLoad { .. } => 1,
             // derivatives use the control flow
             crate::Expression::Derivative { .. } => 1,
+            crate::Expression::Load { pointer } => match expressions[pointer] {
```

Comment:
This is not a complete fix, right? The expression can also be an access into something that is a variable, or it can be a global variable. In all the cases we want to load early.
So technically, returning `1` here is the only good solution. Why are we not doing this?

Reply:
It's a temporary solution. We can safely add Global Variable here without any validation error. I will try to investigate how we can fix glsl array and named expression on wgsl frontend side.

Reply:
What about `Access` and `AccessIndex`?

I'm not sure how you are seeing this being addressed on WGSL frontend. This seems like a backend issue.

Reply:
Wouldn't it be simpler to just return 1 here without checking the actual expressions? I don't understand what this gives us.